com.sap.adt.abapcleaner.gui/src/com/sap/adt/abapcleaner/gui/FrmMain.java

```diff
@@ -185,7 +185,57 @@ public static void cleanAutomatically(CommandLineArgs commandLineArgs, PrintStre
 			}
 		}
 
-		// perform the cleanup
+		// check if single or multi file clean up
+		if (commandLineArgs.sourceFiles != null && commandLineArgs.sourceFiles.length > 0) {
+		  cleanMultiAutomatically(commandLineArgs, out, profile);
+		} else {
+		  cleanSingleSourceAutomatically(commandLineArgs, commandLineArgs.sourceCode, out, profile);
+		}
+	}
+	
+  private static void cleanMultiAutomatically(CommandLineArgs commandLineArgs, PrintStream out,
+      Profile profile) {
+    
+    Persistency persistency = Persistency.get();
+    for (String sourceFile : commandLineArgs.sourceFiles) {
+      String sourceCode = persistency.readAllTextFromFile(sourceFile);
+      
+      CleanupResult result = cleanAutomatically(sourceCode, commandLineArgs.abapRelease, commandLineArgs.cleanupRange, profile, commandLineArgs.showStats);
+      if (result == null) {
+        out.println("Cleanup for file " + sourceFile +" cancelled.");
+        continue;
+      } else if (result.hasErrorMessage()) {
+        out.println("Errors during clean-up of file: " + sourceFile);
+        out.println(result.errorMessage);
+        continue;
+      } 
+
+      // the output is either the whole code document or the cleanup result of the line selection 
+      String output = null;

```

Comment:
everything from here on is very similar to the corresponding section of cleanSingleSourceAutomatically. Maybe both could be extracted to a common method, e.g. writeCleanupResult(CommandLineArgs commandLineArgs, PrintStream out, CleanupResult result, String targetPath)? For the multi-file cleanup, the targetPath can be calculated with persistency.combinaPaths(targetDir, persistency.getFileName(sourcePath)).